core/src/text/language.rs

```diff
@@ -0,0 +1,28 @@
+use crate::text::{Text, Translate};
+
+pub trait Language {
+    fn translate_with<I>(&self, translate: Translate, with: I) -> Text
+    where
+        I: IntoIterator,
+        I::Item: Into<Text>;
+}
+
+impl Language for std::collections::HashMap<Translate, String> {
```

Comment:
Could this be made generic over the hasher parameter `S`? https://rust-lang.github.io/rust-clippy/master/#implicit_hasher